src/reprepro_updater/helpers.py

```diff
@@ -77,18 +77,59 @@ def _run_update_command(repo_dir, distro, commit):
     return try_run_command(update_command)
 
 
+def _get_dependent_packages(repo_dir, distro, arch, package):
+    """ Return a list of packages which were detected as dependant by reprepro.
+    This only returns direct dependencies. """
+
+    reprepro_command = [
+        'reprepro', '-V', '-b', repo_dir,
+        '-T', 'deb',
+        'listfilter', distro,
+        "Package (% ros-* ), " +
+        "Architecture (== " + arch + " ), " +
+        "( Depends (% *" + package + "[, ]* ) " +
+        "| Depends (% *" + package + " ) )"]
+
+    try:
+        output = subprocess.check_output(reprepro_command)
+    except subprocess.CalledProcessError as ex:
+        print "Execution of [%s] Failed:" % reprepro_command, ex
+        return []
+    packages = []
+    for l in output.splitlines():
+        elements = l.split()
+        if len(elements) != 3:
+            print "Expected 3 elements got %s, [%s]" % len(elements), elements
+        packages.append(elements[1])
+    return packages
+
+
```

Comment:
Can you please move this function to the end of the patch? Currently the order of functions is 3, 1, 2 - 1, 2, 3 would be easier to read.